bvm/ballerina-runtime/src/test/java/io/ballerina/runtime/test/cli/OperandTest.java

```diff
@@ -18,13 +18,193 @@
+        Assert.assertTrue(args[1] instanceof BArray);
+        BArray bArray = (BArray) args[1];
+        Assert.assertEquals(bArray.get(0), 1L);
+        Assert.assertTrue(args[3] instanceof Long);
+        Assert.assertEquals(args[3], 2L);
+    }
+
+    @Test
+    public void testIntFollowingArrayWith2Args() {
+        Operand[] operands = {new Operand(false, "arrInt", TypeCreator.createArrayType(PredefinedTypes.TYPE_INT)),
+                new Operand(false, "opString", PredefinedTypes.TYPE_INT)};
+        Object[] args = new CliSpec(null, operands, "1", "2", "3").getMainArgs();
+        Assert.assertTrue(args[1] instanceof BArray);
+        BArray bArray = (BArray) args[1];
+        Assert.assertEquals(bArray.get(0), 1L);
+        Assert.assertEquals(bArray.get(1), 2L);
+        Assert.assertTrue(args[3] instanceof Long);
+        Assert.assertEquals(args[3], 3L);
+    }
+
+    @Test
+    public void testTwoIntsFollowingArray() {
+        Operand[] operands = {new Operand(false, "arrInt", TypeCreator.createArrayType(PredefinedTypes.TYPE_INT)),
+                new Operand(false, "opInt1", PredefinedTypes.TYPE_INT), new Operand(false, "opInt2",
+                                                                                    PredefinedTypes.TYPE_INT)};
+        Object[] args = new CliSpec(null, operands, "1", "2", "3").getMainArgs();
+        Assert.assertTrue(args[1] instanceof BArray);
+        BArray bArray = (BArray) args[1];
+        Assert.assertEquals(bArray.get(0), 1L);
+        Assert.assertTrue(args[3] instanceof Long);
+        Assert.assertEquals(args[3], 2L);
+        Assert.assertTrue(args[5] instanceof Long);
+        Assert.assertEquals(args[5], 3L);
+    }
+
+    @Test
+    public void testOperandIntArrayString() {
+        Operand[] operands = {new Operand(false, "intArr", TypeCreator.createArrayType(PredefinedTypes.TYPE_INT)),
+                new Operand(false, "opString", PredefinedTypes.TYPE_STRING)};
+        String val = "arg";
+        Object[] args = new CliSpec(null, operands, val).getMainArgs();
+        Assert.assertTrue(args[1] instanceof BArray);
+        Assert.assertEquals(((BArray) args[1]).size(), 0);
+        Assert.assertTrue(args[3] instanceof BString);
+        Assert.assertEquals(((BString) args[3]).getValue(), val);
+    }
+
+    @Test
+    public void testStringIntArray() {
+        Operand[] operands = {new Operand(false, "opString", PredefinedTypes.TYPE_STRING),
+                new Operand(false, "arrInt", TypeCreator.createArrayType(PredefinedTypes.TYPE_INT))};
+        String val = "arg";
+        Object[] args = new CliSpec(null, operands, val).getMainArgs();
+        Assert.assertTrue(args[1] instanceof BString);
+        Assert.assertEquals(((BString) args[1]).getValue(), val);
+        Assert.assertTrue(args[3] instanceof BArray);
+        Assert.assertEquals(((BArray) args[3]).size(), 0);
+    }
+
+    @Test
+    public void testStringUnionNil() {
```

Comment:
Do we have tests for rest params with value and without values?

Reply:
Added tests as jballerina-unit-tests